pinot-spi/src/main/java/org/apache/pinot/spi/trace/Tracing.java

```diff
@@ -272,8 +272,12 @@ public static void initializeThreadAccountant(PinotConfiguration config) {
           LOGGER.warn("ThreadAccountant {} register unsuccessful, as it is already registered.", factoryName);
         }
       } catch (Exception exception) {
-        LOGGER.warn("Using default implementation of thread accountant, "
-            + "due to invalid thread accountant factory {} provided, exception:", factoryName, exception);
+        if (factoryName != null) {
```

Comment:
Should we do the null check only only construct the `ThreadAccountantFactory` when `factoryName != null`?